### PR TITLE
Update AddOneAsync to return modified struct and checks to RaceResult POST

### PR DESCRIPTION
**Pull request by MutatedGamer**

### CODE REVIEW

File: api/tests/Data/Utils/MockContainerProvider.cs [hunk 1]
```diff
@@ -60,6 +60,8 @@ public static Container CreateMockContainer(List<T> data)
                     });
 
             container.GetItemLinqQueryable<T>().Returns(data.AsQueryable());
+            
+            container.GetItemLinqQueryable<T>(Arg.Any<bool>()).Returns(data.AsQueryable());
```
Comment (luillo1) [inline api/tests/Data/Utils/MockContainerProvider.cs:64]: Why is this needed?

Comment (MutatedGamer) [reply api/tests/Data/Utils/MockContainerProvider.cs:64]: The clients are calling GetItemLinqQueryable(true), which is returning null since that method isn't mocked

Comment (MutatedGamer) [reply api/tests/Data/Utils/MockContainerProvider.cs:64]: I believe some tests fail on main atm

**Review by luillo1 — APPROVED** [review]
